### ISSUE TITLE
Support alt text as short caption for Latex

### ISSUE THREAD

**Issue by hadim** [lierdakil/pandoc-crossref#38]
Using this patch (https://github.com/jgm/pandoc/pull/2447) allow to use image alt text to convert in latex short caption. When it is used with pandoc-crossref, the short caption becomes empty...

See :

```markdown
![My long caption](figure.png "SHORT CAPTION"){#fig:figure-label}
```

produces

```latex
\begin{figure}[htbp]
\centering
\includegraphics{figure.png}
\caption[]{\label{fig:figure-label}My long caption}
\end{figure}
```

Instead of : 

```latex
\begin{figure}[htbp]
\centering
\includegraphics{figure.png}
\caption[SHORT CAPTION]{\label{fig:figure-label}My long caption}
\end{figure}
```

Any idea to fix it ?

**Comment by lierdakil** [comment]
With latex output, pandoc-crossref writes raw latex. So to fix this, one
needs to patch pandoc-crossref. I will try to take a look later this week.
12 окт. 2015 г. 14:50 пользователь "Hadrien Mary" <notifications@github.com>
написал:

**Comment by hadim** [comment]
I have the feeling there is something to do there : https://github.com/lierdakil/pandoc-crossref/blob/a8110d81cb780f84f4f5d52a45b14ac896a359a8/src/Text/Pandoc/CrossRef/References/Blocks.hs#L39 but I suck too much in Haskell to be able to fix it. If you guide me I can maybe try to propose a patch.

If you think it involves a bit of work, I'll let you do it later this week.

**Comment by lierdakil** [comment]
At first glance line 41 should probably have "fig:" ++ snd img instead of
"fig:".
12 окт. 2015 г. 15:46 пользователь "Hadrien Mary" <notifications@github.com>
написал:

**Comment by hadim** [comment]
`return $ Para [Image alt' (fst img, "fig:" ++ snd img)]` produces the following : 

```latex
\begin{figure}[htbp]
\centering
\includegraphics{figure.png}
\caption[fig:]{\label{fig:figure-label}My long caption}
\end{figure}
```

and `return $ Para [Image alt' (fst img, "fig:" ++ "test" ++ snd img)]` produces:

```latex
\begin{figure}[htbp]
\centering
\includegraphics{figure.png}
\caption[testfig:]{\label{fig:figure-label}My long caption}
\end{figure}
```

Seems weird to me... is haskell an inverted language ?

**Comment by lierdakil** [comment]
That's probably Pandoc, not Haskell. I don't have immediate access to
Pandoc's source ATM, so I'm not sure what's going on. Note that Pandoc's
document model uses "fig:" prefix for implicit figures, so it probably
shouldn't end up in LaTeX output...
12 окт. 2015 г. 16:15 пользователь "Hadrien Mary" <notifications@github.com>
написал:

**Comment by hadim** [comment]
Ok. Thanks anyway. I will wait for a fix then !

**Comment by lierdakil** [comment]
Ok, so 9ef29f971feb5314f8e55b75dae860b5dcee0c70 should fix this. Sorry about misleading you, I kinda forgot my own code :sweat_smile:  Would be great if you could test with  jgm/pandoc#2447 before I publish a release.

**Comment by hadim** [comment]
It works like a charm ! Thank you.

However there is a little issue and I can't tell wether it is related to pandoc-crossref or pandoc.

String inside short caption does not seems to be parsed by pandoc. For example a short caption (inside `[]`) such as `test _italic_` is not rendered with italic style but as `test _italic_`.

Any idea ?

**Comment by hadim** [comment]
After some tests, it seems related to pandoc. I will report on https://github.com/jgm/pandoc/pull/2447

Thanks again @lierdakil !

**Comment by lierdakil** [comment]
Yep, that's expected. Pandoc doesn't parse image title as markdown (I think that's primarily due to "title" being an html attribute, which can't have rich formatting)

**Comment by hadim** [comment]
Ok I guess I can live with that :-)

I close

**Comment by lierdakil** [comment]
FYI, 9ef29f9 published as 0.1.5.5
